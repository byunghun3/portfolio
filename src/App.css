.App {
  text-align: center;
  scroll-behavior: smooth;
}

.sectionsContainer {
  height: 100vh;
  scroll-snap-type: y mandatory;
  /* overflow-y: scroll; */
}

.sectionOne {
  height: 100vh;
  background-color: red;
  /* scroll-snap-align: start; */
}

.sectionTwo {
  height: 100vh;
  background-color: blue;
  scroll-snap-align: start;
}

.sectionThree {
  position: relative;
  height: 200vh;
  background-color: green;
  scroll-snap-align: start;
}

.scroll {
  position: absolute;
}