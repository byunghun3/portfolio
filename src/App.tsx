import React, { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { Profile } from "./components/Profile";
import { Projects } from "./components/Projects";
import { Story } from "./components/Story";
import { ThemeProvider } from "./contexts/ThemeContext";
import "./App.css";
import styled from "styled-components";

const Container = styled.div`
  height: 100vh;
  width: 100%;
  scroll-snap-type: y mandatory;
  overflow-y: scroll;

  @media (max-height: 599px), (max-width: 600px) {
    // min-height: 100vh;
    // display: block;
    scroll-snap-type: none;
    overflow-y: visible;
    // background-color: purple;
  }

  // @media (max-width: 499px) {
  //   min-height: 100vh;
    // scroll-snap-type: none;
    // overflow-y: visible;
  // }
`;

const Section = styled.div`
      display: flex;
    flex-direction: column;
    // min-height: 100vh;
  // scroll-snap-align: start;
`;

function App() {
  return (
    <ThemeProvider>
      <div className="App">
        <Container>
          {/* <Section id="profile"> */}
          {/* a */}
          <Profile
            projectsHash="#projects"
          />
          {/* </Section> */}
          {/* <Section id="projects"> */}
          {/* b */}
          <Projects
            profileHash="#profile"
          />
          {/* </Section> */}
          {/* <Section id="story"> */}
          <Story
            profileHash="#profile"
          />
          {/* </Section> */}
        </Container>
      </div>
    </ThemeProvider>
  );
}

export default App;