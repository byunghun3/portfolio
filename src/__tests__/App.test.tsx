import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../App";

const MockApp = () => {
    return (
        <BrowserRouter>
            <App />
        </BrowserRouter>
    );
};

test("anchor tags match corresponding links", () => {
    render(<MockApp />);

    // const imageElement = screen.getByAltText("profile-picture");
    // expect(imageElement).toBeInTheDocument();
    // const projectsButtonElement = screen.getByRole("button", { name: "Projects" });
    // const storyButtonElement = screen.getByRole("button", { name: "Story" });
    // fireEvent.click(projectsButtonElement);
    // fireEvent.click(storyButtonElement);
    // expect(window.location.pathname).toEqual("/#projects");
    // expect(window.location.pathname).toEqual("/#story");
    const resumeDemoLinkElement = screen.getByLabelText("Resume");
    const linkedInDemoLinkElement = screen.getByLabelText("LinkedIn");
    const githubLinkElement = screen.getByLabelText("Github");
    const bookstoreDemoLinkElement = screen.getByLabelText("Revd Bookstore Demo");
    const bookstoreCodeLinkElement = screen.getByLabelText("Revd Bookstore Code");
    const golfDemoLinkElement = screen.getByLabelText("Golf Course Weather Demo");
    const golfCodeLinkElement = screen.getByLabelText("Golf Course Weather Code");
    const portfolioDemoLinkElement = screen.getByLabelText("Portfolio Demo");
    const portfolioCodeLinkElement = screen.getByLabelText("Portfolio Code");
    // fireEvent.click(bookstoreDemoLinkElement);
    // expect(window.location.pathname).toEqual("https://byunghun3.github.io/revd-bookstore/");
    expect(resumeDemoLinkElement).toHaveAttribute("href", "resume.pdf");
    expect(linkedInDemoLinkElement).toHaveAttribute("href", "https://www.linkedin.com/in/byung-hun-kim-a3809a143/");
    expect(githubLinkElement).toHaveAttribute("href", "https://github.com/byunghun3");
    expect(bookstoreDemoLinkElement).toHaveAttribute("href", "https://byunghun3.github.io/revd-bookstore/");
    expect(bookstoreCodeLinkElement).toHaveAttribute("href", "https://github.com/byunghun3/revd-bookstore");
    expect(golfDemoLinkElement).toHaveAttribute("href", "https://byunghun3.github.io/golf-course-weather/");
    expect(golfCodeLinkElement).toHaveAttribute("href", "https://github.com/byunghun3/golf-course-weather");
    expect(portfolioDemoLinkElement).toHaveAttribute("href", "https://byunghun3.github.io/byunghun3/portfolio");
    expect(portfolioCodeLinkElement).toHaveAttribute("href", "https://github.com/byunghun3/portfolio");
});