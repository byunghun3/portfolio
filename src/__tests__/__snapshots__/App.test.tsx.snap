// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link: /#projects 1`] = `"/"`;
